assignments/arrays.js

```diff
@@ -75,30 +75,91 @@ let inventory = [
 
 // ==== Challenge 1 ====
 // The dealer can't recall the information for a car with an id of 33 on his lot. Help the dealer find out which car has an id of 33 by logging the car's year, make, and model in the console log provided to you below:
-console.log(`Car 33 is a *car year goes here* *car make goes here* *car model goes here*`);
+
+
+  for(i = 0; i < inventory.length; i++){
+    if(inventory[i].id === 33){
+      console.log("Car 33 is a " + inventory[i].car_year + " " + inventory[i].car_make + " " + inventory[i].car_model + ".");
+    }
+  }
+
+
 
 // ==== Challenge 2 ====
 // The dealer needs the information on the last car in their inventory.  What is the make and model of the last car in the inventory?  Log the make and model into the console.
+// function lastCar(inventory){
+//   let lastCar = 0;
+//   for(i = lastCar; i < inventory.length; i++){
+//     if(inventory[i].id === inventory.length)
+//     {
+//       return `${inventory[i].car_make}, ${inventory[i].car_model}`
+//     }
+//   }
+// }
+// console.log(lastCar(inventory));
 let lastCar = 0;
-console.log();
+
+  for (let i = 0; i < inventory.length; i++){
+      if (inventory[i].id === 50) {
+     console.log(inventory[i].car_make + ' ' + inventory[i].car_model)
+      }// did not use lastCar ask TL
+  }
+
 
 // ==== Challenge 3 ====
 // The marketing team wants the car models listed alphabetically on the website. Sort all the car model names into alphabetical order and log the results in the console
+// function sortModels(inventory){
+// let carModels = [];
+// for(i = 0; i < inventory.length; i++){
+  
+//   carModels.push(inventory[i].car_model);
+// }
+//   return carModels.sort();
+// }
+// console.log(sortModels(inventory));
+
 let carModels = [];
-let carModelsSorted = [];
-console.log();
+
+    for (let i = 0; i < inventory.length; i++){
+    let models = inventory[i].car_model;
+    carModels.push(models);
+    carModels.sort();
```

Comment:
As discussed, take the sort method outside of the for loop. You want to sort it after the entire array is filled up and not while it's filling up.